scripts/ai/AIDriveStrategyCourse.lua

```diff
@@ -512,11 +504,16 @@ function AIDriveStrategyCourse:unhold()
         self:debug("Hold reset")
     end
     self.held:reset()
+    self.fuelSaveActiveWhileHeld = false
 end
 
 --- Are we currently being held?
 function AIDriveStrategyCourse:isBeingHeld()
-    return self.held:get()
+    local isHeld = self.held:get()
+    if not isHeld then 
+        self.fuelSaveActiveWhileHeld = false

```

Comment:
This isn't very clean, one would not expect a query function like this to have side effects.